src/contracts/acs.ts

```diff
@@ -18,23 +15,21 @@ export class AnyoneCanSpend extends SmartContract {
     @method()
     public unlock(txPreimage: SigHashPreimage, outputAmount: bigint) {
 
-        assert(this.checkPreimageSigHashType(txPreimage, SigHash.ANYONECANPAY_SINGLE));
-
-        let lockingScript: string = Utils.buildPublicKeyHashScript(this.pubKeyHash);
-        let output: string = Utils.buildOutput(lockingScript, outputAmount);
-        assert(hash256(output) == SigHash.hashOutputs(txPreimage));
+        assert(this.checkPreimageSigHashType(txPreimage, SigHash.ANYONECANPAY_SINGLE), 'preimage check failed');
```

Comment:
Use this syntax and remove `checkPreimage*` everywhere.
https://github.com/sCrypt-Inc/scrypt-ts/blob/4c45b5ded1a8ae625e8d57ebad6211bc570b0d4c/test/contracts/acs.ts#L17